src/weather/tests/weather.service.spec.ts

```diff
@@ -0,0 +1,139 @@
+    status,
+    statusText: 'Bad Request',
+    headers: null,
+    config: null,
+    data: {
+      statusCode: status,
+      error: 'Bad Request',
+      message: {},
+    },
+  });
+
+  const requestCorrectResponse = (data: any): AxiosResponse => ({
+    status: 200,
+    statusText: 'Success',
+    headers: null,
+    config: null,
+    data,
+  });
+
+  beforeAll(async (): Promise<void> => {
+    const module: TestingModule = await Test.createTestingModule({
+      providers: [
+        WeatherService,
+        {
+          provide: HttpService,
+          useValue: {
+            get: jest.fn().mockImplementation(() => of()),
+          },
+        },
+        {
+          provide: ConfigService,
+          useValue: {
+            get: jest.fn(() => 'dummy_api_key'),
+          },
+        },
+      ],
+    }).compile();
+
+    httpService = module.get(HttpService);
+    weatherService = module.get<WeatherService>(WeatherService);
+  });
+
+  it('should be defined', () => {
+    expect(weatherService).toBeDefined();
+  });
+
+  describe('getForecast', () => {
+    // it's no longer necessary once there is class validator
+
+    // const invalidValues: number[] = [undefined, null, NaN];
+    //
+    // for (const lat of invalidValues) {
+    //   it(`should throw BadRequestException for ${lat} as latitude`, async (): Promise<void> => {
+    //     const expectedError = new BadRequestException(
+    //       WEATHER_ERROR.INVALID_LATITUDE_ERROR,
+    //       '400',
+    //     );
+    //
+    //     const response = weatherService.getForecast(lat, 10);
+    //
+    //     await expect(response).rejects.toThrow(expectedError);
+    //     await expect(response).rejects.toBe(
+    //       WEATHER_ERROR.INVALID_LATITUDE_ERROR,
+    //     );
+    //   });
+    // }
+    //
+    // for (const lng of invalidValues) {
+    //   it(`should throw BadRequestException for ${lng} as longitude`, async (): Promise<void> => {
+    //     const expectedError = new BadRequestException(
+    //       WEATHER_ERROR.INVALID_LONGITUDE_ERROR,
+    //       '400',
+    //     );
+    //
+    //     await expect(weatherService.getForecast(10, lng)).rejects.toThrow(
+    //       expectedError,
+    //     );
+    //   });
+    // }
+
+    it('should throw BadRequestException for response error', async (): Promise<void> => {
+      const mockResponse = new BadRequestException(
+        {
+          ...WEATHER_ERROR.RESPONSE_ERROR,
+          error: {},
+        },
+        '400',
+      );
+      jest.spyOn(httpService, 'get').mockImplementation(
+        () =>
+          new Observable(
+            (observer: Subscriber<AxiosResponse<any, any>>): void => {
+              observer.error({
+                response: requestErrorResponse(),
+                config: null,
+                isAxiosError: true,
+                name: '',
+                message: 'error message',
+                toJSON: jest.fn(),
+              });
+            },
+          ),
```

Comment:
I believe there is a shorthand for this called `throwError` from `rxjs` package

```typescript
import { of, throwError } from 'rxjs';

// return observable with some data (successful)
if (success) {
  return of({ /* some response */ });
} else {
  return throwError(() => /* some error response */ });
}
```

In your case, something like this
```typescript
throwError(() => ({
  response: requestErrorResponse(),
  config: null,
  isAxiosError: true,
  name: "",
  message: "error message",
  toJSON: jest.fn(),
}));
```